spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -763,6 +764,35 @@ public void testDefaultSortOrder() {
     assertEquals("Data after compaction should not change", expectedRecords, actualRecords);
   }
 
+  @Test
+  public void testRewriteDataFilesWithSystemFunctions() {
+    Assumptions.assumeThat(catalogName).isNotEqualTo("spark_catalog");
+
+    sql(
+        "CREATE TABLE %s (c1 INT, c2 STRING, c3 TIMESTAMP) "
+            + "USING iceberg "
+            + "PARTITIONED BY (days(c3)) "
+            + "TBLPROPERTIES ('%s' '%s')",
+        tableName,
+        TableProperties.WRITE_DISTRIBUTION_MODE,
+        TableProperties.WRITE_DISTRIBUTION_MODE_NONE);
+
+    sql(
+        "INSERT INTO TABLE %s VALUES (0, 'data-0', CAST('2017-11-22T09:20:44.294658+00:00' AS TIMESTAMP))",
+        tableName);
+    sql(
+        "INSERT INTO TABLE %s VALUES (1, 'data-1', CAST('2017-11-23T03:15:32.194356+00:00' AS TIMESTAMP))",
+        tableName);
+    // Test with invalid filter column col1
+    Assertions.assertThatThrownBy(
+            () ->
+                sql(
+                    "CALL %s.system.rewrite_data_files(table => '%s', where => '%s.system.years(c3) = 2017')",
```

Comment:
looks like comment suggest filter on column col1 but predicate in where clause is actually on year transform of c3, just want to make sure if it's desired

Reply:
Thanks for the good catch. I need to update to the UT since we will support the system function filters with #8560

Reply:
UT updated.